### ISSUE TITLE
is there any way conditionally trigger MenuItem selection event?

### ISSUE THREAD

**Issue by kante-srikanth** [ericgio/react-bootstrap-typeahead#812]
Use case: I am using the renderMenu prop and MenuItem component to render the custom UI. In the below code example let's say I have two buttons Delete and Share resp. I want to perform different actions based on the button clicked.

code ex:

```
const renderMenu = (results, menuProps, state) => (
  results.map(result => (
     <MenuItem>
         <button>Delete</button>
         <button>Share</button>
     </MenuItem>
  ))
)
```

<TypeAhead renderMenu={renderMenuHandler} />

Expectation: When clicking on the Delete button I want to show a different UI and want the MenuItem selection event not to be fired.
Actual: It is firing button click event and MenuItem selection event.

Basically, I want to conditionally trigger the MenuItem onClick event. 

Is there any efficient way of doing this? Thanks

**Comment by ericgio** [comment]
@kante-srikanth: Yes, you can stop propagation on the delete button.

**Comment by tapizquent** [comment]
@ericgio how would you do this? I am hoping to do something similar where I need a custom event when selecting an option (or rather a custom button I added to the bottom of the list)

**Comment by ericgio** [comment]
```tsx
<MenuItem>
  <button onClick={(e) => e.stopPropagation()}>Some action</button>
</MenuItem>
```

**Comment by duhmojo** [comment]
I just opened this: https://github.com/ericgio/react-bootstrap-typeahead/issues/842

You can set disabled in an option item, but it doesn't seem to be used for anything with renderMenu. You can however access the option.disabled and pass it into MenuItem. e.g. disabled your MenuItem. This will stop the highlighting and make the item appear dead and unselectable. 

```
<MenuItem key={idx} option={o} position={idx} disabled={ o.disabled }>{o.label}</MenuItem>
```

However, just like you I'm trying to add some input/button elements to a menu/option item at the parent menu wrapper is blocking access. Even if its disabled, the wrapper seems to be focusing on the Typeahead area at the top when you click any menu item and the click isn't making it to the input, or it's redirected by the focus. I'm trying to figure out how to stop this on disabled menu items myself.
